lib/hls/hls_parser.js

```diff
@@ -888,6 +888,29 @@ shaka.hls.HlsParser = class {
     return count;
   }
 
+  /**
+   * Get the spatial audio information for an HLS audio track.
+   * The MP4 registration authority (mp4ra.org) lists a value of ec+3 for
+   * Enhanced AC-3 audio with JOC (Dolby Atmos). That value is not used by HLS.
+   * Instead, it uses ec-3 and marks the presence of the additional JOC
+   * content with JOC in the CHANNELS attribute of the audio rendition. The JOC
+   * MUST be capitalized. For example, CHANNELS="16/JOC". The numeric value
+   * should match the value of the complexity_index_type_a field in the
+   * EC3SpecificBox of the Dolby Digital Plus audio track.
+   * @see https://developer.apple.com/documentation/http_live_streaming/hls_authoring_specification_for_apple_devices/hls_authoring_specification_for_apple_devices_appendixes
```

Comment:
Sorry, I read this three times, but still couldn't make anything of it :/
Would you mind explaining the relation between spatial audio and HLS to me in a more human-friendly language :p 
Here is what I think I read:
The JOC info is stored under the "CHANNELS" field. What's up with the "value not used by HLS" part? And what's with the complexity_index_type?

Reply:
In HLS the channels field indicates the number of audio channels that the stream has (eg: 2). In the case of Dolby Atmos, the complexity is expressed with the number of channels followed by the word JOC (eg: 16/JOC), so 16 would be the number of channels (eg: 7.3.6 layout), and JOC indicates that the stream has spatial audio.

Reply:
Understood. Could I ask you to replace the spec copy-paste part with this explanation, and just keep the reference to the spec at the end?
Like this:
Get the spatial audio information for an HLS audio track.
In HLS the channels field indicates the number of audio channels that the stream has (eg: 2). In the case of Dolby Atmos, the complexity is expressed with the number of channels followed by the word JOC (eg: 16/JOC), so 16 would be the number of channels (eg: 7.3.6 layout), and JOC indicates that the stream has spatial audio.
see HLS doc

Reply:
Changed!